### ISSUE TITLE
Component scope attribute is lost when conditionally rendering root node

### ISSUE THREAD

**Issue by Frizi** [vuejs/vue#8485]
### Version
2.5.16

### Reproduction link
[https://codesandbox.io/s/5vj19q8yk](https://codesandbox.io/s/5vj19q8yk)

### Steps to reproduce
1. observe that text is green
2. click two times on checkbox to trigger slot's hide and show cycle

### What is expected?
3. text should still be green

In other words, the toggled `div` should have `GreenSlot`'s `data-v` attribute applied

### What is actually happening?
3. observe that text is red

In other words, the toggled `div` doesn't have correct `data-v` applied.

---
Also tested on beta, issue persists.

In my use-case, I cannot trivially replace it with external `v-if`, as the component that does the toggling contains important internal logic that decides if it should be shown or not. It's not just a simple prop.

**Comment by posva** [comment]
As a workaround make sure to always have a node for the ConditionalSlot component (<div v-else/>)
But I'm not sure it should be green actually, why would the style of the wrapper component leak into the slot that was written in the parent?

**Comment by Frizi** [comment]
That's because the `<slot>` is also a part of component, so it receives it's scope. In the case from the example, it ends up being the `div` with "I should be green" text directly. In fact, this `div` is a node present in all `App`, `GreenSlot` and `ConditionalSlot` components at the same time.
